system/modules/core/library/Contao/Validator.php

```diff
@@ -388,4 +388,35 @@ public static function isInsecurePath($strPath)
 
 		return false;
 	}
+
+
+	/**
+	 * Valid file name
+	 *
+	 * @param mixed $strName The file name
+	 *
+	 * @return boolean True if the file name is valid
+	 */
+	public static function isValidFileName($strName)
+	{
+		// Special characters not supported on e.g. Windows
+		if (preg_match('@[\\\\/:*?"<>\|]@', $strName))
+		{
+			return false;
+		}
+
+		// Invisible control characters or unused code points
+		if (preg_match('/[\pC]/u', $strName))
+		{
+			return false;
+		}
+
+		// Must not be longer than 255 characters
+		if (utf8_strlen($strName) > 255)
```

Comment:
I think this test should be extended to

```php
// Must not be longer than 255 characters but at least 1 character
$length = utf8_strlen($strName);
if ($length > 255 || $length < 1)
```

Otherwise an empty string would be a valid file name.